spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarReactiveProperties.java

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.autoconfigure;
+
+import java.time.Duration;
+
+import org.apache.pulsar.client.api.CompressionType;
+import org.apache.pulsar.client.api.HashingScheme;
+import org.apache.pulsar.client.api.MessageRoutingMode;
+import org.apache.pulsar.client.api.ProducerAccessMode;
+import org.apache.pulsar.client.api.ProducerCryptoFailureAction;
+import org.apache.pulsar.reactive.client.api.MutableReactiveMessageSenderSpec;
+import org.apache.pulsar.reactive.client.api.ReactiveMessageSenderSpec;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.PropertyMapper;
+
+/**
+ * Configuration properties for Spring for the Apache Pulsar reactive client.
+ * <p>
+ * Users should refer to Pulsar reactive client documentation for complete descriptions of
+ * these properties.
+ *
+ * @author Christophe Bornet
+ */
+@ConfigurationProperties(prefix = "spring.pulsar.reactive")
+public class PulsarReactiveProperties {
+
+	private final Sender sender = new Sender();
+
+	public Sender getSender() {
+		return this.sender;
+	}
+
+	public ReactiveMessageSenderSpec buildReactiveMessageSenderSpec() {
+		return this.sender.buildReactiveMessageSenderSpec();
+	}
+
+	public static class Sender {
+
+		/**
+		 * Topic the producer will publish to.
+		 */
+		private String topicName;
+
+		/**
+		 * Name for the producer. If not assigned, a unique name is generated.
+		 */
+		private String producerName;
+
+		/**
+		 * Time before a message has to be acknowledged by the broker in milliseconds.
+		 */
+		private Duration sendTimeout = Duration.ofSeconds(30);
+
+		/**
+		 * Maximum number of pending messages for the producer.
+		 */
+		private int maxPendingMessages = 1000;
```

Comment:
Typically, it is better to use big primitives (Integer vs. int) for config props values if you want to allow null (aka no setting). That being said, the existing PP do not follow that so we can revisit this later for both flavors of the props.

Reply:
done